docker-images/kafka-based/kafka/scripts/set_kafka_jmx_options.sh

```diff
@@ -6,7 +6,7 @@ JMX_USERNAME="$2"
 JMX_PASSWORD="$3"
 
 if [ "$JMX_ENABLED" = "true" ]; then
-  KAFKA_JMX_OPTS="-Dcom.sun.management.jmxremote.port=9999 -Dcom.sun.management.jmxremote.rmi.port=9999 -Dcom.sun.management.jmxremote=true -Dcom.sun.management.jmxremote.ssl=false -Djava.rmi.server.hostname=$(hostname -i) -Djava.net.preferIPv4Stack=true"
+  KAFKA_JMX_OPTS="-Dcom.sun.management.jmxremote.port=9999 -Dcom.sun.management.jmxremote.rmi.port=9999 -Dcom.sun.management.jmxremote=true -Dcom.sun.management.jmxremote.ssl=false -Djava.rmi.server.hostname=$(hostname -I | awk '{print $1}') -Djava.net.preferIPv4Stack=true"
```

Comment:
`-I` results in the ipv4 being first, `-i` puts ipv6 first.

i could go back to `-i` and parse the ipv4 out, but i thought this is cleaner

Reply:
Why does the IPv4 being used matter if you have dual-stack networking? I think the `awk` part works good. But parsing it in some more sophisticated way or changing the `-i` to -I` makes a big risk that it breaks things for someone else as the output of these things is a bit unpredictable.

Reply:
sorry i didnt explain myself, when the pod has dual stack networking, heres how it looks:
```bash
[kafka@sonic-cluster-kafka-0 kafka]$ hostname -i
2600:1900:4000:deee:0:1f:0:1b 10.216.0.27
[kafka@sonic-cluster-kafka-0 kafka]$ hostname -I
10.216.0.27 2600:1900:4000:deee:0:1f:0:1b 
[kafka@sonic-cluster-kafka-0 kafka]$ 
```
and when running on a single stack pod:
```bash
root@we# hostname -I
10.212.6.50 
root@we# hostname -i
10.212.6.50
```

Reply:
so something like this would be better?
`hostname -i | grep -oE '\b([0-9]{1,3}\.){3}[0-9]{1,3}\b'`

Reply:
> so something like this would be better? `hostname -i | grep -oE '\b([0-9]{1,3}\.){3}[0-9]{1,3}\b'`

As I said, I think `hostname -I | awk '{print $1}'` is probably better than that. But `hostname -i | awk '{print $1}'` should IMHO work  as well and is even better.

Reply:
its the opposite, with `awk '{print $1}'` im getting the ipv4 and ignoring the ipv6, the problem might not have anything to do with ipv6 but instead just that `-Djava.rmi.server.hostname` expects a single value and not multiple.
so `hostname -I` and then taking the first result should be what we want

Reply:
That is my assumption as well that it needs a single value. But `hostname -i | awk '{print $1}' gives you a single value and makes sure that for everyone for whom it worked today it will keep working even after this change because it will use the same `hostname -i` command,

Reply:
Or let me put it differently ... what does not work for you on `hostname -i | awk '{print $1}'`?

Reply:
i can try it, but im pretty sure it wont work, since im using ipv4 to connect to jmx. and using `hostname -i | awk '{print $1}'` will result in it binding to the ipv6

Reply:
And why would it not work over IPv6?

Reply:
And how do you know the IPv4 will be the first address? Is it documented anywhere?

Reply:
its not. should i rather use grep to parse the ip out?

Reply:
I guess let's try it as it is and see how it goes.